boa/src/syntax/parser/statement/iteration/do_while_statement.rs

```diff
@@ -63,53 +63,31 @@ where
 
     fn parse(self, cursor: &mut Cursor<R>) -> Result<Self::Output, ParseError> {
         let _timer = BoaProfiler::global().start_event("DoWhileStatement", "Parsing");
-        cursor.expect(Keyword::Do, "do while statement", false)?;
+        cursor.expect(Keyword::Do, "do while statement")?;
 
         let body =
             Statement::new(self.allow_yield, self.allow_await, self.allow_return).parse(cursor)?;
 
-        let next_token = cursor.peek(0, true)?.ok_or(ParseError::AbruptEnd)?;
+        let next_token = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?;
 
         if next_token.kind() != &TokenKind::Keyword(Keyword::While) {
             return Err(ParseError::expected(
                 vec![TokenKind::Keyword(Keyword::While)],
-                next_token,
+                next_token.clone(),
                 "do while statement",
             ));
         }
 
-        cursor.expect(Keyword::While, "do while statement", true)?;
+        cursor.expect(Keyword::While, "do while statement")?;
 
-        cursor.expect(Punctuator::OpenParen, "do while statement", true)?;
+        cursor.expect(Punctuator::OpenParen, "do while statement")?;
 
         let cond = Expression::new(true, self.allow_yield, self.allow_await).parse(cursor)?;
 
-        cursor.expect(Punctuator::CloseParen, "do while statement", true)?;
+        cursor.expect(Punctuator::CloseParen, "do while statement")?;
 
-        expect_semicolon_dowhile(cursor)?;
+        cursor.expect_semicolon(true, "do while statement")?;
 
         Ok(DoWhileLoop::new(body, cond))
     }
 }
-/// Checks that the next token is a semicolon with regards to the automatic semicolon insertion rules

```

Comment:
Surely if this is a unique case to do_while it should stay within the do_while file?

Reply:
Hmmm you are right, that would reduce a bit the overhead in the rest of the checks.